classic_startup/python2.7libs/pythonrc.py

```diff
@@ -43,6 +44,26 @@ def classic_startup():
     # add to the system path
     sys.path.insert(0, tk_houdini_python_path)
 
+    # We need to check to make sure we don't have an incompatibility
+    # between httplib and Houdini's bundled ssl.py. This is a problem
+    # on some Linux distros (CentOS 7.x) with H16.
+    if sys.platform == "linux2" and sys.version.startswith("2.7.5"):
+        # We can check to see if ssl has the function we know that
+        # system httplib is likely to require. If it doesn't have it,
+        # then we need to force the use of our bundled httplib before
+        # we let the bootstrap happen.
+        import ssl
+        if not hasattr(ssl, "_create_default_https_context"):
+            # Clear httplib if it's already been imported.
+            if "httplib" in sys.modules:
+                del sys.modules["httplib"]
+
+            # Add the submodule containing httplib to sys.path so that
+            # the next time it's imported it'll come from there instead
+            # of the system Python install.
+            httplib_path = os.path.join(tk_houdini_python_path, "packages")
+            sys.path.insert(0, httplib_path)
```

Comment:
I'm a little worried about deleting from `sys.modules` without replacing it. Would it be better to manually repopulate here to be safe?